atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/service/WriteBatchingTransactionServiceTest.java

```diff
@@ -129,12 +130,15 @@ public void throwsOnUnspecifiedKeyAlreadyExistsExceptions() {
 
     @Test
     public void repeatedProcessBatchDoesNotBreakOnItsOwnSuccessfulWrites() {
-        KeyAlreadyExistsException originalException = new KeyAlreadyExistsException("boo", ImmutableList.of(
-                ENCODING_STRATEGY.encodeStartTimestampAsCell(3L)));
-        KeyAlreadyExistsException newExistingKeyException = new KeyAlreadyExistsException("boo", ImmutableList.of(
-                ENCODING_STRATEGY.encodeStartTimestampAsCell(2L)));
-
-        doThrow(originalException).doThrow(originalException)
+        KeyAlreadyExistsException originalException = new KeyAlreadyExistsException(
+                "boo",
+                ImmutableList.of(ENCODING_STRATEGY.encodeStartTimestampAsCell(3L)),
+                ImmutableList.of(ENCODING_STRATEGY.encodeStartTimestampAsCell(2L)));
+        KeyAlreadyExistsException newExistingKeyException = new KeyAlreadyExistsException(
```

Comment:
This is now never going to be thrown because the list of timestamps will be empty (once success, one failure). Should probably add a third timestamp that neither succeeded nor failed on the first attempt and verify we retry for that one

Reply:
Yep, makes sense - I added a third `elementToRetry`